### PR TITLE
`#![no_std]`, embedded support

### LINKED ISSUE #243: Can't build with no_std because of cfg-if dependency
```
 └─  cargo c
    Updating crates.io index
       F       Fetch [================>                                      ]  31.39%                                                                                                        Downloaded time-macros v0.1.0
  Downloaded time-macros-impl v0.1.0
  Downloaded standback v0.2.2
  Downloaded proc-macro-hack v0.5.15
  Downloaded time v0.2.9
  Downloaded 5 crates (123.9 KB) in 0.91s
   Compiling syn v1.0.17
   Compiling rustversion v1.0.2
   Compiling standback v0.2.2
   Compiling proc-macro-hack v0.5.15
    Checking cfg-if v0.1.10
error[E0463]: can't find crate for `std`
  |
  = note: the `thumbv7em-none-eabihf` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `standback`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

### PR DESCRIPTION
**Pull request by PTaylor-FluenTech**
Fixes #243 
Requires jhpratt/standback#6

This should allow `time` to be used with no_std targets **assuming** the author of standback accepts my PR.

The second commit isn't required, but (in my opinion) is a cleaner way of checking the `std` feature. It might be completely wrong.

### CODE REVIEW

**Comment by PTaylor-FluenTech** [comment]
Waiting for resolution of jhpratt/standback#6 and version bump.

File: time-macros/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![no_std]
```
Comment (PTaylor-FluenTech) [inline time-macros/src/lib.rs:1]: I believe this is ok even when this crate is used by other projects that are not no_std.

**Comment by jhpratt** [comment]
Standback is my crate, FYI. Consider that done after a quick review later.

The second commit is correct in functionality, but I felt it was cleaner to emit two separate flags. I'm open to being convinced, though!

As with standback, I'll handle the CI side of things to avoid unnecessary duplication.

File: time-macros/src/lib.rs (hunk 1 above)
Comment (jhpratt) [reply time-macros/src/lib.rs:1]: That's correct. I'll check the semantics on whether this is necessary, as it's only present at build time.

**Comment by PTaylor-FluenTech** [comment]
> Standback is my crate, FYI.

Boy do I feel silly 😳.

>The second commit is correct in functionality, but I felt it was cleaner to emit two separate flags. I'm open to being convinced, though!

As I've mentioned, I'm pretty new. I think I've seen it as in my commit more, but don't have strong feelings about it. I think for me, it doesn't _hide_ the `no_std` flag. When I see `cfg(std)`, I know where that `std` configuration is coming from just by looking at the `Cargo.toml`. Just my two cents. Thank you for confirming that it's functionally correct.

**Comment by jhpratt** [comment]
> Boy do I feel silly :flushed:.

No worries! Unless you were looking at who has control on crates.io or who has committed to the repos, there's no reason for you to know. And realistically, it doesn't much matter.

> As I've mentioned, I'm pretty new. I think I've seen it as in my commit more, but don't have strong feelings about it. I think for me, it doesn't _hide_ the `no_std` flag. When I see `cfg(std)`, I know where that `std` configuration is coming from just by looking at the `Cargo.toml`. Just my two cents. Thank you for confirming that it's functionally correct.

For what it's worth if it's clearer to new people, that's typically what's preferred! cfg aliases certainly isn't something a beginner needs to be familiar with, either.

File: time-macros/src/lib.rs (hunk 1 above)
Comment (jhpratt) [reply time-macros/src/lib.rs:1]: Looks like it is necessary, as it's not a proc_macro crate.

**Comment by jhpratt** [comment]
Looks like everything is working as expected. All I really did was get CI to a place I like.

Thank you!

**Pull request merged**